server/src/main/java/com/microsoft/java/bs/core/internal/services/LifecycleService.java

```diff
@@ -119,61 +134,117 @@ public void exit() {
   enum Status {
     UNINITIALIZED,
     INITIALIZED,
-    SHUTDOWN;
+    SHUTDOWN
   }
 
   /**
-   * Try to update the Gradle Java home if:
-   * <ul>
-   *   <li>Gradle Java home is not set.</li>
-   *   <li>A valid JDK can be found to launch Gradle.</li>
-   * </ul>
-   *
-   * <p>The JDK installation path string will be set to {@link Preferences#gradleJavaHome}.
+   * Finds a suitable version of JDK to use for compilation of Java programs.
```

Comment:
Let's say a JDK compatible with the target Gradle version. Because it's the jdk which is used for Gradle operations (for example running tasks or acquiring model information). It's not only for compilation.

Reply:
Addressed in commit [fbf550b](https://github.com/microsoft/build-server-for-gradle/pull/165/commits/fbf550b288559f5f92a474307885ce56e417d962)